### DISCUSSION TITLE
[BUG] Abort and Exit break autocompletion

### DISCUSSION THREAD

**Discussion by cauebs** [fastapi/typer#1583] [Questions] [answered]
I believe these two types of exceptions should be handled internally to just abort the autocompletion (equivalent to returning an empty list), instead of leaking and flooding the terminal with a traceback.

I'm willing to try and fix this if maintainers agree :)

**Comment by xmnlab** [comment]
I have the same issue here.
any workaround for that?

**Comment by xmnlab** [comment]
for my case, my command can pass a parameter where is the config file `--file`.
the default is `.makim.yaml`

so if there is no `.makim.yaml` in the current directory, the command fails.

but with autocompletion .. if there is no `.makim.yaml`, it should just not show anything ... 
if the user want to do something like `makim --file [TAB][TAB]` it is currently failing

**Comment by YuriiMotov** [comment, marked as answer]
While I agree that we should try handling this automatically, the workaround that should help is described in [Fix completion - using the Context](https://typer.tiangolo.com/tutorial/options/callback-and-context/#fix-completion-using-the-context)

**Comment by mpouzoup** [comment]
Hey, I'd like to work on this issue if it's still unassigned!
